maas/resource_maas_network_interface_vlan.go

```diff
@@ -167,7 +167,7 @@ func resourceNetworkInterfaceVlanUpdate(ctx context.Context, d *schema.ResourceD
 		return diag.FromErr(err)
 	}
 
-	vlan, err := getVlan(client, fabric.ID, d.Get("vlan").(string))
+	vlan, err := getVlan(client, fabric.ID, fmt.Sprintf("%v", d.Get("vlan").(int)))
```

Comment:
What is the benefit over previous approach here?
`fmt.Sprintf("%v", d.Get("vlan").(int))` vs `d.Get("vlan").(string)`

Reply:
https://pkg.go.dev/strconv#Itoa

Reply:
This is actually the fix for: `panic: interface conversion: interface {} is int, not string`. Good catch on the less performant usage of `fmt.Sprintf`. Let me update both create and update functions to use Itoa